ext/standard/file.c

```diff
@@ -1831,8 +1831,8 @@ PHP_FUNCTION(fputcsv)
 	if (delimiter_str != NULL) {
 		/* Make sure that there is at least one character in string */
 		if (delimiter_str_len < 1) {
-			php_error_docref(NULL, E_WARNING, "delimiter must be a character");
-			RETURN_FALSE;
+			zend_value_error("delimiter must be a character");
+			return;
 		} else if (delimiter_str_len > 1) {
 			php_error_docref(NULL, E_NOTICE, "delimiter must be a single character");
```

Comment:
I think it's important that we keep these pairs of warnings symmetrical. Either both should throw or neither.

Reply:
Symmetrical with the SPL ones right? I'll do the changes at the same time borrowing parts from #4991

Reply:
Symmetrical with the error above. I now realize though that in this case we throw a warning but still continue running the function, ugh. I think @cmb69's suggestion above to make this a single check (with BC break) may make the most sense.

Reply:
Ah ok, want me to make that part of this PR or seperate one?